mm2src/mm2_bitcoin/serialization/src/reader.rs

```diff
@@ -57,29 +57,40 @@ pub enum CoinVariant {
     LBC,
     Standard,
     PPC,
+    /// Needed to deserialize `RICK` block headers correctly as it's genesis header is `version 1` while all other
+    /// block headers are `version 4`.
+    RICK,
+    /// Same reason as RICK.
+    MORTY,
 }
 
 impl CoinVariant {
     pub fn is_btc(&self) -> bool { matches!(self, CoinVariant::BTC) }
-
+    pub fn is_btc_or_ppc(&self) -> bool { matches!(self, CoinVariant::BTC | CoinVariant::PPC) }
     pub fn is_qtum(&self) -> bool { matches!(self, CoinVariant::Qtum) }
-
     pub fn is_lbc(&self) -> bool { matches!(self, CoinVariant::LBC) }
-
-    pub fn is_ppc(&self) -> bool { matches!(self, CoinVariant::PPC) }
+    pub fn is_morty_or_rick(&self) -> bool { matches!(self, CoinVariant::RICK | CoinVariant::MORTY) }
 }
 
 impl From<&str> for CoinVariant {
     fn from(ticker: &str) -> Self {
+        let match_variant = |ticker: &str, with| {
+            ticker == with || ticker.contains(&format!("{with}-")) || ticker.contains(&format!("{with}_"))
+        };
```

Comment:
Closure may capture a local variable, so you don't need to take `ticker`:
```rust
let match_variant = |with| {
            ticker == with || ticker.contains(&format!("{with}-")) || ticker.contains(&format!("{with}_"))
        };
```

Otherwise you could add a general function
```rust
fn ticker_matches(ticker: &str, with: &str) -> bool { todo!() }
```

Reply:
done